cmd/server/main.go

```diff
@@ -5,10 +5,116 @@
 package main
 
 import (
+	"context"
+	"crypto/tls"
+	"flag"
 	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/moov-io/base/admin"
+	moovhttp "github.com/moov-io/base/http"
+	"github.com/moov-io/base/http/bind"
 	"github.com/moov-io/wire"
+
+	"github.com/go-kit/kit/log"
+	"github.com/gorilla/mux"
+)
+
+var (
+	httpAddr  = flag.String("http.addr", bind.HTTP("fed"), "HTTP listen address")
+	adminAddr = flag.String("admin.addr", bind.Admin("fed"), "Admin HTTP listen address")
+
+	flagLogFormat = flag.String("log.format", "", "Format for log lines (Options: json, plain")
 )
 
 func main() {
-	fmt.Printf("Starting moov-io/wire version %s\n", wire.Version)
+	flag.Parse()
+
+	var logger log.Logger
+	if strings.ToLower(*flagLogFormat) == "json" {
+		logger = log.NewJSONLogger(os.Stderr)
+	} else {
+		logger = log.NewLogfmtLogger(os.Stderr)
+	}
+	logger = log.With(logger, "ts", log.DefaultTimestampUTC)
+	logger = log.With(logger, "caller", log.DefaultCaller)
+
+	logger.Log("startup", fmt.Sprintf("Starting fed server version %s", wire.Version))
+
+	// Channel for errors
+	errs := make(chan error)
+
+	go func() {
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+		errs <- fmt.Errorf("%s", <-c)
+	}()
+
+	// Start Admin server (with Prometheus metrics)
+	adminServer := admin.NewServer(*adminAddr)
+	go func() {
+		logger.Log("admin", fmt.Sprintf("listening on %s", adminServer.BindAddr()))
+		if err := adminServer.Listen(); err != nil {
+			err = fmt.Errorf("problem starting admin http: %v", err)
+			logger.Log("admin", err)
+			errs <- err
+		}
+	}()
+	defer adminServer.Shutdown()
+
+	// Setup business HTTP routes
+	router := mux.NewRouter()
+	moovhttp.AddCORSHandler(router)
+	addPingRoute(router)
```

Comment:
Do we need to add the file routes onto this Router?

Reply:
Not in its current format.